### PR TITLE
Feature: new FlexibleEFTHiggs up to 3L in full-model parametrization

### PR DESCRIPTION
**Pull request by Expander**
Implements the shooting method for FlexibleEFTHiggs models. This shooting method enables the inclusion of 2- and 3-loop contributions to the light Higgs pole mass in FlexibleEFTHiggs models.

This PR allows to generate the 3-loop FlexibleEFTHiggs calculation of the light Higgs pole mass in the (real) MSSM from https://arxiv.org/abs/2003.04639 . In addition it allows to generate a 2-loop FlexibleEFTHiggs calculation of light Higgs pole mass in the NMSSM (new).

### CODE REVIEW

**Comment by Expander** [comment]
I believe this PR is read to be merged.

**Review by wkotlarski — APPROVED** [review]
Please have a look at indentation. I think some places mix spaces and tabs. Other than that I'm ok with all your changes. You know what you're doing :+1:

File: meta/FlexibleSUSY.m [hunk 1]
```diff
@@ -1445,7 +1454,15 @@ corresponding tadpole is real or imaginary (only in models with CP
             setRunningDownLeptonMasses = "", setYukawas = "",
             calculateMUpQuarkPole1L = "", calculateMDownQuarkPole1L = "",
             calculateMDownLeptonPole1L = "",
-            calculateMHiggsPoleOneMomentumIteration = ""},
+            calculateMHiggsPoleNoMomentumIteration = "",
+	    calculateMHiggs2LoopShift = "throw SetupError(\"2-loop Higgs self-energy not enabled.\");",
```
[the remark this answers is not in the document]
Comment (Expander) [reply meta/FlexibleSUSY.m]: fixed in 1a8f59a07

**Comment by Expander** [comment]
I think I have to re-visit this PR, because we have changed the calculation of `g1` and `g2` in the SM (for good reason).

**Comment by Expander** [comment]
@pathron @wkotlarski This PR is ready to be merged. I nobody objects, I'll merge it after tomorrow.

**Review by wkotlarski — APPROVED** [review]

**Review by pathron — APPROVED** [review]
Sorry I have not reviewed it in detail  (so many lines of code) but i think it can be merged since Wojtek approved it.  It sounds like really nice progress!

**Pull request merged**